### PR TITLE
Pre-pend SK search path in error messages

### LINKED ISSUE #870: SK search path error
**Is your feature request related to a problem? Please describe.**
Search path in use for SK files is not clear if the code cannot find anything. This is printed as relative to the DFTBPLUS_PARAM_DIR

**Describe the solution you'd like**
If set, DFTBPLUS_PARAM_DIR should be pre-pended to where the search failed, or an initial message that it is not set before the relative path for the SK files.

### PR DESCRIPTION
**Pull request by vanderhe**
Had to stop working on it (needs discussion). The current version assumes that the maximum number of search paths is one, not as trivial as I thought.

Eventually closes #870.

### CODE REVIEW

**Comment by vanderhe** [comment]
@aradi Does this work for you?

**Review by aradi — COMMENTED** [review]
The solution with `endsWidth()` is very nice, I like it. :wink: 

However, probably we should make `findFile()` and the entire mechanism more consistent:

* `findFile()` should always return an unallocated char, when the file was not found (also for absolut path). That means, after having calling `findFile`, no `inquire()` is needed any more. If it returns an allocated char, the file with that name can be opened. If it returns an unallocated, no file can be opened.
* `findFile()` should probably not look in the current folder, but in the ones in `searchPath`. (We should then add `./` element to the `searchPath` when calling `findFile()`.
* When generating error message due to not-found SK-files: We should print the name of the file (e.g. `C-C.skf`) and then separately the paths, where it had been searched for:
  ```
  SK-file 'Pu-Pu-skf' not found (search paths: './', '/opt/slakos/mio-1-1').
  ```
  This should provide precise information about what happened.

**Comment by vanderhe** [comment]
@aradi Should address your comments now, but the hard coded OS dependent relative CWD path in `getParamSearchPaths()` is not nice.

**Review by aradi — APPROVED** [review]

File: src/dftbp/dftbplus/parser.F90 [hunk 1]
```diff
@@ -1390,16 +1391,14 @@ subroutine readDFTBHam(node, ctrl, geo, slako, poisson, errStatus)
...
-          call append(skFiles(iSp2, iSp1), strTmp)
-          inquire(file=strTmp, exist=tExist)
-          if (.not. tExist) then
-            call detailedError(value1, "SK file with generated name '" &
-                &// trim(strTmp) // "' does not exist.")
+          if (.not. allocated(strOut)) then
+            call detailedError(value1, "SK file with generated name '" // trim(strTmp)&
+                & // "' not found." // newline // "   (search path(s): " // strJoin // ").")
           end if
+          strTmp = strOut
```
[the remark this answers is not in the document]
Comment (vanderhe) [reply src/dftbp/dftbplus/parser.F90:1400]: @aradi Hard coded './' changed to '.'.

**Pull request merged**